### ISSUE TITLE
WOC_WRITE_ONLY_COLLECTION_LOCAL false positive if collection is read in lambda

### ISSUE THREAD

**Issue by kilotaras** [mebigfatguy/fb-contrib#39]
If data is read from collections in lambda WOC_WRITE_ONLY_COLLECTION_LOCAL is generated.

**Comment by kilotaras** [comment]
sample code

```` java

import java.util.ArrayList;
import java.util.List;
import java.util.stream.IntStream;

public class FbFail {
    public String fail() {
        List<String> l = new ArrayList<>();

        l.add("0");
        l.add("1");
        l.add("2");
        l.add("3");
        l.add("4");

        return IntStream.of(1,2,3).mapToObj(i -> l.get(i)).findFirst().get();
    }

    public String dontFail() {
        List<String> l = new ArrayList<>();

        l.add("0");
        l.add("1");
        l.add("2");
        l.add("3");
        l.add("4");

        return IntStream.of(1,2,3).mapToObj(l::get).findFirst().get();
    }
}

```

**Comment by mebigfatguy** [comment]
Thanks for the report. Fixed in 0f50f6afa8e6c493d9dc91152017e4c07c807631, due to missing handling of INVOKEDYNAMIC
